lib/open_food_network/feature_toggle.rb

```diff
@@ -1,21 +1,78 @@
 module OpenFoodNetwork
+  # This feature toggles implementation provides two mechanisms to conditionally enable features.
+  #
+  # You can provide an ENV var with the prefix `OFN_FEATURE_` and query it using the
+  # `ApplicationHelper#feature?` helper method. For instance, providing the ENV var
+  # `OFN_FEATURE_NEW_SHINNY_FEATURE` you could then query it from view as follows:
+  #
+  #   - if feature? :new_shinny_feature
+  #     = render "new_shinny_feature"
+  #
+  # Alternatively, you can choose which users have the feature toggled on. To do that you need to
+  # register the feature and its users from an initializer like:
+  #
+  #   require 'open_food_network/feature_toggle'
+  #   OpenFoodNetwork::FeatureToggle.enable(:new_shinny_feature, ['ofn@example.com'])
+  #
+  # It's although advisable to read the emails list from an ENV Var.
+  #
+  # You can then check it from a view like:
+  #
+  #   - if feature? :new_shinny_feature, spree_current_user
+  #     = render "new_shinny_feature"
+  #
   class FeatureToggle
-    def self.enabled?(feature_name)
-      true?(env_variable_value(feature_name))
+    def self.enabled?(feature_name, user = nil)
+      new.enabled?(feature_name, user)
+    end
+
+    def self.enable(feature_name, user_emails)
+      Thread.current[:features] ||= {}
+      Thread.current[:features][feature_name] = Feature.new(:foo, user_emails)
```

Comment:
the fact that you are using foo here is a smell that indicates, imho, that these should be to separate classes, NamedFeatureToggle and UserBaseFeatureToggle or something similar.
I dont think this is important enough, that's why I approve the PR, but this solution does trigger interesting OOP problems.

Reply:
By `NamedFeatureToggle` you mean separating the ENV-based approach from this new one? I thought so too but my goal is to get the simplest thing possible out the door, use it and see how we need it to evolve. I don't even know if others the team like this idea.